awx/main/signals.py

```diff
@@ -637,6 +640,25 @@ def delete_inventory_for_org(sender, instance, **kwargs):
             logger.debug(e)
 
 
+@receiver(pre_delete, sender=WorkflowJobTemplateNode)
+def delete_approval_nodes(sender, instance, **kwargs):
+    if type(instance.unified_job_template) is WorkflowApprovalTemplate:
```

Comment:
Another signal I could see us adding is one that detects `WorkflowApprovalTemplate` deletions and automatically deletes any pending associated `WorkflowApproval`s.  Something like:

```python
@receiver(post_delete, sender=WorkflowJobApprovalTemplate)
def delete_approval_nodes(sender, instance, **kwargs):
    for approval in WorkflowApproval.objects.filter(workflow_approval_template=instance, status='pending'):
        approval.deny()
```
```